classes/Kohana/Route.php

```
@@ -473,10 +480,12 @@ public function matches(Request $request)
 				{
 					// Filter has modified the parameters
 					$params = $return;
+					// @todo if $return != $params, then uppercase values (directory, controller)
 				}
 			}
 		}
 
+		// @todo throw exception then some params not exists (controller, action)
```

My mistake, assumed it was but you're right so yes we should throw here if required params are missing.